controllers/subscription/subscription_controller.go

```diff
@@ -246,40 +258,53 @@ func (r *SubscriptionReconciler) HandleUpgrades(ctx context.Context, rhmiSubscri
 		}
 	}
 
-	isInstallPlanDeleted := false
-	currentOperatorVersionName := fmt.Sprintf("%s.v%s", CSVNamePrefix, version.GetVersion())
+	csvNamePrefix := "integreatly-operator"
+	if installation.Spec.Type == string(integreatlyv1alpha1.InstallationTypeManagedApi) {
+		csvNamePrefix = "managed-api-service"
+	}
+
+	currentOperatorVersionName := fmt.Sprintf("%s.v%s", csvNamePrefix, version.GetVersion())
 	if csvFromCatalogSource.Spec.Replaces != currentOperatorVersionName {
+		err = rhmiConfigs.DeleteInstallPlan(ctx, latestRHMIInstallPlan, r.Client)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("error deleting installplan %w", err)
+		}
 
-		if csvFromCatalogSource.Spec.Replaces != latestRHMICSV.Spec.Replaces {
-			err = rhmiConfigs.DeleteInstallPlan(ctx, latestRHMIInstallPlan, r.Client)
-			if err != nil {
-				return ctrl.Result{}, fmt.Errorf("error deleting installplan %w", err)
-			}
+		log.Info("Installplan deleted for the install of patch upgrade")
 
-			isInstallPlanDeleted = true
-			log.Info("Installplan deleted for the install of patch upgrade")
+		err = rhmiConfigs.CreateInstallPlan(ctx, rhmiSubscription, r.Client)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
 
-			err = rhmiConfigs.CreateInstallPlan(ctx, rhmiSubscription, r.Client)
+		err := wait.Poll(time.Second*5, time.Minute*1, func() (done bool, err error) {
+			// gets the subscription with the recreated installplan
+			logrus.Infof("Entered wait.Poll")
+			err = r.Client.Get(ctx, k8sclient.ObjectKey{Name: rhmiSubscription.Name, Namespace: rhmiSubscription.Namespace}, rhmiSubscription)
+			logrus.Infof("got subscription")
 			if err != nil {
-				return ctrl.Result{}, err
+				log.Infof("Couldn't get subscription", l.Fields{"Error": err})
+				return false, nil
 			}
-		}
-
-		if isInstallPlanDeleted {
-			// Requeue reconciler until the installplan is recreated
-			return ctrl.Result{
-				Requeue:      true,
-				RequeueAfter: 10 * time.Second,
-			}, nil
-		}
-
-		// gets the subscription wit the recreated installplan
-		err := r.Client.Get(ctx, k8sclient.ObjectKey{Name: rhmiSubscription.Name, Namespace: rhmiSubscription.Namespace}, rhmiSubscription)
+			logrus.Infof("InstallPlan read from the new subscription")
+			if rhmiSubscription.Status.InstallPlanRef == nil {
+				log.Info("Install plan was not referenced in the subscription")
+				return false, nil
+			}
+			latestRHMIInstallPlan, err = rhmiConfigs.GetLatestInstallPlan(ctx, rhmiSubscription, r.Client)
+			if err != nil {
+				log.Info("Install plan was not created")
+				return false, nil
+			}
+			logrus.Infof("new install plan %s", latestRHMIInstallPlan.Name)
+			return true, nil
+		})
 		if err != nil {
+			log.Info("reconcile loop")
```

Comment:
I'd change this message to say "Triggering a new reconcile loop due to an error after waiting for the creation of a new install plan" and the `err` variable to the log